docker/importer/importer.py

```diff
@@ -701,42 +703,77 @@ def _process_deletions_bucket(self,
                          self._public_log_bucket, import_failure_logs)
 
   def _process_updates_rest(self, source_repo: osv.SourceRepository):
-    """Process updates from REST API."""
+    """Process updates from REST API.
+    
+    To find new updates, first makes a HEAD request to check the 'Last-Modified'
+    header, and skips processing if it's before the source's last_modified_date
+    (and ignore_last_import_time isn't set).
+
+    Otherwise, GETs the list of vulnerabilities and requests updates for
+    vulnerabilities modified after last_modified_date.
+
+    last_modified_date is updated to the HEAD's 'Last-Modified' time, or the
+    latest vulnerability's modified date if 'Last-Modified' was missing/invalid.
+    """
     logging.info('Begin processing REST: %s', source_repo.name)
 
-    ignore_last_import_time = (
-        source_repo.ignore_last_import_time or not source_repo.last_update_date)
-    if ignore_last_import_time:
+    last_update_date = source_repo.last_update_date or datetime.datetime.min
+    if source_repo.ignore_last_import_time:
+      last_update_date = datetime.datetime.min
       source_repo.ignore_last_import_time = False
       source_repo.put()
-    import_time_now = utcnow()
-    request = requests.head(source_repo.rest_api_url, timeout=_TIMEOUT_SECONDS)
+
+    s = requests.Session()
+    adapter = HTTPAdapter(
+        max_retries=Retry(
+            total=3, status_forcelist=[502, 503, 504], backoff_factor=1))
+    s.mount('http://', adapter)
```

Comment:
Should we be supporting http (not s) endpoints?

Reply:
The tests use http (although I haven't actually added a test for the retries). I don't think there's any harm in supporting it here regardless.